src/sources/opentelemetry/http.rs

```diff
@@ -59,6 +68,8 @@ pub(crate) async fn run_http_server(
 
     let span = Span::current();
     let make_svc = make_service_fn(move |conn: &MaybeTlsIncomingStream<TcpStream>| {
+        let remote_addr = conn.peer_addr();
+        let remote_addr_ref = enable_source_ip.then_some(remote_addr);
```

Comment:
The following is preferable because it's `Some` only `enable_source_ip` is `true`:
```suggestion
 let remote_addr = enable_source_ip.then(|| conn.peer_addr());
```

Reply:
Thanks, if I come back to this work to add `host_key` to `opentelemetry` I will do this. I had copied what the `http_server` source did.